nni/algorithms/compression/pytorch/pruning/one_shot_pruner.py

```diff
@@ -0,0 +1,174 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import logging
+
+import torch
+from nni.common.graph_utils import TorchModuleGraph
+from nni.compression.pytorch.compressor import Pruner
+from nni.compression.pytorch.utils.config_validation import CompressorSchema
+from nni.compression.pytorch.utils.shape_dependency import (ChannelDependency,
+                                                            GroupDependency)
+from schema import And, Optional, SchemaError
+from .dependency_aware_pruner import DependencyAwarePruner
+
+__all__ = ['LevelPruner', 'L1FilterPruner', 'L2FilterPruner', 'FPGMPruner']
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+class OneshotPruner(DependencyAwarePruner):
```

Comment:
Why does the `OneshotPruner` inherit from the `DependencyAwarePruner`? Shouldn't the `DependencyAwarePruner` be inherited from OneShotPruner?  Not all one shot pruners can be dependency-aware(level-pruner).

Reply:
Is it possible to achieve dependency-aware for level-pruner even though it is hard to implement?

This may because both iterative pruners (slim, Taylor...) and one-shot pruner(l1, l2..) can support dependency-aware.
My plan is to add not implementation error or warning for those un-supported pruners.

Reply:
Got it.

Reply:
Level-pruner is basically impossible to achieve dependency-aware.

Reply:
Got it, and seems we only have level pruner as a special pruner that hard to achieve dependency-aware? maybe we can detach level pruner from inheriting `DependencyAwarePruner` in follow-up pr.

Reply:
It sounds good to me after your kindly explanation. No need to separate the level-pruner specifically. Please go ahead.